src/server/editorServices.ts

```diff
@@ -272,14 +272,29 @@ module ts.server {
 
     export class Project {
         compilerService: CompilerService;
-        projectOptions: ProjectOptions;
         projectFilename: string;
         program: ts.Program;
         filenameToSourceFile: ts.Map<ts.SourceFile> = {};
         updateGraphSeq = 0;
+        /** Used for configured projects which may have multiple open roots */
+        openRefCount = 0;
 
-        constructor(public projectService: ProjectService) {
-            this.compilerService = new CompilerService(this);
+        constructor(public projectService: ProjectService, public projectOptions?: ProjectOptions) {
+            if (projectOptions && projectOptions.compilerOptions) {
+                this.compilerService = new CompilerService(this,projectOptions.compilerOptions);
+            }
+            else {

```

Comment:
i would make it `this.compilerService = new CompilerService(this, projectOptions && projectOptions.compilerOptions); `